ReferencePluginF/ControlF.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using System.Xml.Serialization;
+
+using Paratext.PluginInterfaces;
+
+
+namespace ReferencePluginF
+{
+	public partial class ControlF : EmbeddedPluginControl
+	{
+		public const string savedDataId = "PluginFData.xml";
+		public const string xmlRoot = "ExtraProjectData";
+
+		private string m_currentText;
+		private string m_savedText;
+		private IWriteLock m_writeLock;
+		private IProject m_project;
+
+
+		private static readonly XmlSerializer m_Serializer = new XmlSerializer(typeof(ProjectTextData));
+
+		public ControlF()
+		{
+			InitializeComponent();
+			m_currentText = null;
+			m_savedText = null;
+			m_writeLock = null;
+			m_project = null;
+		}
+
+		public override void OnAddedToParent(IPluginChildWindow parent, IWindowPluginHost host, string state)
+		{
+			m_project = parent.CurrentState.Project;
+			parent.SetTitle(PluginF.pluginName);
+			parent.SaveRequested += SaveRequested;
+			parent.WindowClosing += WindowClosing;
+			parent.ProjectChanged += ProjectChanged;
+		}
+
+		public override string GetState()
+		{
+			return null;
+		}
+
+		public override void DoLoad(IProgressInfo progressInfo)
+		{
+			// Actually do the load:
+			// Since DoLoad is done on a different thread than what was used
+			// to create the control, we need to use the Invoke method.
+			Invoke((Action)(() => LoadSavedText()));
+
+			// Illustrate use of progressInfo
+			const int MAX_TIME = 100;
+			progressInfo.Initialize("Loading Reference F", MAX_TIME);
+			for (int i = 0; i < MAX_TIME; i++)
+			{
+				if (progressInfo.CancelRequested)
+				{
+					m_writeLock.Dispose();
+					m_writeLock = null;
+					textBox.Text = "";
+					textBox.ReadOnly = true;
+				}
+				Thread.Sleep(20);
+				progressInfo.Value = i;
+			}
+		}
+
+		private void ReleaseRequested(IWriteLock writeLock)
+		{
+			writeLock.Dispose();
+			m_writeLock = null;
+			textBox.ReadOnly = true;
+		}
+
+		private void LoadSavedText()
+		{
+			if (m_project == null)
+			{
+				MessageBox.Show("Project not provided");
+				m_savedText = "";
+				m_currentText = "";
+				textBox.Text = "";
+				textBox.ReadOnly = true;
+			}
+			else
+			{
+				m_writeLock = m_project.RequestWriteLock(this, ReleaseRequested, savedDataId);
```

Comment:
In the case where the reload button was clicked, the existing writelock might not have been disposed. You should only need/want to try to get a new lock if m_writeLock is null. Otherwise, it will always fail and return null and you won't successfully reload (since you abort loading below in that case).